doc/stdlib/c.rst

```diff
@@ -209,10 +262,35 @@
 
    Raises a ``SystemError`` for ``errno`` with the descriptive string ``sysfunc`` if ``bool`` is true
 
-.. function:: strerror(n)
+.. function:: strerror(errno)
 
    Convert a system call error code to a descriptive string
 
+.. data:: Ptr{T}
+
+   A simple pointer to an arbitrary memory location.
+   The type objects expected at the memory location is represented by the type parameter.
+   However, no guarantee is made or implied that the memory is actually valid,
+   or that it actually represents the data of the specified type.
+
+   ``C_NULL`` represents a generic, invalid or "NULL" pointer.
+
+.. data:: Ref{T}
+
+   Effectively, this represents and creates a managed pointer.
+   The type of objects it can contain is specified by the type parameter.
+   This type is guaranteed to point to valid, Julia-allocated memory
+   of the correct type (per the type parameter).
+
+   When passed to a `ccall` argument (either as a `Ptr` or `Ref` type),
+   the `Ref` object will be implicitly converted to a pointer to the
+   data region of that type.
+
+   The ``Ref`` type is useful for creating garbage-collector safe pointers and
+   returning values from a function (esp. a c-function), for example.
```

Comment:
Does "c-function" mean a function written in C, or a julia function passed to C with `cfunction`?